assets/src/edit-story/components/panels/textStyle/padding.js

```diff
@@ -52,7 +52,9 @@ function PaddingControls({ selectedElements, pushUpdateForObject }) {
     DEFAULT_PADDING
   );
 
-  const [lockPaddingRatio, setLockPaddingRatio] = useState(true);
+  const [lockPaddingRatio, setLockPaddingRatio] = useState(
+    padding.horizontal === padding.vertical
+  );
```

Comment:
It's better to do this in a `useEffect`. This will not rerun if the panel is re-rendered without being unmounted first. For instance, when having one text box selected and then selecting another text box with a different initial padding.

```js
  const [lockPaddingRatio, setLockPaddingRatio] = useState(true);
  useEffect(() => {
    setLockPaddingRatio(padding.horizontal === padding.vertical);
  }, [padding]);
```

Reply:
@barklund In that case, the padding lock will update if the user input same padding on horizontal and vertical while editing without refresh or reselect

Reply:
hmm... yes that's maybe the case of one issue. But if we use useEffect then we have another issue. Like opened padding lock and adjust horizontal and vertical same value then the padding lock will be enabled.

Reply:
But it makes sense for me. And I think that will be the correct UX behavior.